kernel/proc.c

```diff
@@ -431,25 +438,57 @@ scheduler(void)
     intr_on();
     intr_off();
 
-    int found = 0;
+    struct proc *runnable_procs[NPROC];
+    uint32 tickets[NPROC];
+    uint32 total_tickets = 0;
+    int runnable_count = 0;
+
     for(p = proc; p < &proc[NPROC]; p++) {
       acquire(&p->lock);
       if(p->state == RUNNABLE) {
-        // Switch to chosen process.  It is the process's job
-        // to release its lock and then reacquire it
-        // before jumping back to us.
-        p->state = RUNNING;
-        c->proc = p;
-        swtch(&c->context, &p->context);
-
-        // Process is done running for now.
-        // It should have changed its p->state before coming back.
-        c->proc = 0;
-        found = 1;
+        runnable_procs[runnable_count] = p;
+        tickets[runnable_count] = p->tickets;
+        total_tickets += p->tickets;
+        runnable_count++;
       }
       release(&p->lock);
     }
-    if(found == 0) {
+    
+    // 프로세스 A, B, C가 가진 티켓의 수량이 30개, 20개 10개 이면
+    // A: 1~30, B: 31~50, C: 51~60 으로 보정하여 추첨 할 수 있도록
+    // 현재까지 추첨되지 않은 프로세스의 티켓들을 누적
+    uint32 winning_number = rand_range(1, total_tickets);
+    uint32 cumulative_tickets = 0;
+    int found = 0;
```

Comment:
사실 C에도 boolean이 있습니다
```suggestion
    _Bool found = 0;
```